### PR TITLE
Implementing task completion date

### PR DESCRIPTION
**Pull request by hcassus**

### CODE REVIEW

File: src/main/resources/static/app.js [hunk 1]
```diff
@@ -1,123 +1,123 @@
...
-          if (uuid == store.tasks[i].uuid) {
-            store.tasks[i].status = data.status;
-          }
+    var app = angular.module('hrp', [])
+        .filter('titleCase', toTitleCase)
+        .filter('pluralize', pluralize);
+
+    function pluralize() {
+        return function (word, count) {
+            if (count == 1 || count == -1) {
```
Comment (ronanrodrigo) [inline src/main/resources/static/app.js]: I suggest to remove the magic numbers. Or, extract this logic to a `var` like
```js
var isSingularWord = count == 1 || count == -1
```

File: src/main/resources/static/app.js [hunk 2]
```diff
@@ -1,123 +1,123 @@
...
+                    if (uuid == store.tasks[i].uuid) {
+                        store.tasks[i] = data;
+                    }
+                }
+            })
+        };
+
+    }]);
+
+    app.controller('PantryCtrl', ['$http', function ($http) {
```
Comment (ronanrodrigo) [inline src/main/resources/static/app.js]: What means `Ctrl`?

File: src/main/resources/static/app.js [hunk 3]
```diff
@@ -1,123 +1,123 @@
...
+                    store.newItem.unit = data.unit || 'UNKNOWN';
+                }
+            });
+        };
+
+        this.addItem = function () {
+            $http.post(this.pantryUrl, this.newItem).success(function (data) {
+                store.content.push(data);
+            });
+            this.newItem = {'unit': 'UNKNOWN'}
```
Comment (ronanrodrigo) [inline src/main/resources/static/app.js]: Why `'UNKNOWN'`?

File: src/main/resources/static/app.js [hunk 4]
```diff
@@ -1,123 +1,123 @@
...
+        };
+
+        this.removeItem = function (item) {
+            uuid = item.uuid;
+            urlWithId = this.pantryUrl + "/" + uuid;
+            $http.delete(urlWithId).success(function () {
+                                                for (i = 0; i < store.content.length; i++) {
+                                                    if (uuid == store.content[i].uuid) {
+                                                        store.content.splice(i, 1);
+                                                    }
```
Comment (ronanrodrigo) [inline src/main/resources/static/app.js]: I do not understand what is happening here. I suggest to extract in a function

File: src/test/java/tasks/api/TasksApiTest.java [hunk 5]
```diff
@@ -66,9 +71,13 @@ public void getAllTasksTest(){
...
           .body("get(0).status", is(equalTo(task.getStatus())))
           .body("get(0).uuid", is(equalTo(task.getUuid().toString())))
+          .body("get(0).updatedAt", notNullValue(Timestamp.class))
+          .body("get(0).createdAt", notNullValue(Timestamp.class))
           .body("get(1).content", is(equalTo(task2.getDescription())))
           .body("get(1).status", is(equalTo(task2.getStatus())))
-          .body("get(1).uuid", is(equalTo(task2.getUuid().toString())));
+          .body("get(1).uuid", is(equalTo(task2.getUuid().toString())))
+          .body("get(1).updatedAt", notNullValue(Timestamp.class))
+          .body("get(1).createdAt", notNullValue(Timestamp.class));
```
Comment (ronanrodrigo) [inline src/test/java/tasks/api/TasksApiTest.java]: Magic numbers :(

File: src/test/java/tasks/api/TasksApiTest.java [hunk 6]
```diff
@@ -105,32 +114,37 @@ public void completeTask() throws JSONException {
...
+          .body("updatedAt", response -> greaterThan(response.path("createdAt")));
 
     assertThat(repository.count(), is(equalTo(1L)));
   }
 
   @Test
   public void uncompleteTask() throws JSONException {
     Task task = new Task("Task " + System.currentTimeMillis());
-    task.setStatus(0);
+    task.setStatus(1);
```
Comment (ronanrodrigo) [inline src/test/java/tasks/api/TasksApiTest.java]: Magic number :(

File: src/test/java/tasks/api/TasksApiTest.java [hunk 7]
```diff
@@ -105,32 +114,37 @@ public void completeTask() throws JSONException {
...
         .when()
-        .patch("/task/" + task.getUuid())
+          .patch("/task/" + task.getUuid())
         .then()
-        .statusCode(200)
-        .body("status", is(equalTo(0)))
-        .body("uuid", notNullValue())
-        .body("description", is(equalTo(task.getDescription())));
-
+          .statusCode(200)
```
Comment (ronanrodrigo) [inline src/test/java/tasks/api/TasksApiTest.java]: I saw magic numbers everywhere 😨

File: src/test/java/tasks/persistence/TaskGatewayTest.java [hunk 8]
```diff
@@ -67,9 +69,12 @@ public void completeTasks(){
     Task task = new Task("Task " + System.currentTimeMillis());
     UUID uuid = task.getUuid();
     repository.save(task);
+
     gateway.changeTaskStatus(uuid, 1);
 
-    assertThat(repository.findOneByUuid(uuid).getStatus(), is(equalTo(1)));
+    Task persistedTask = repository.findOneByUuid(uuid);
+    assertThat(persistedTask.getStatus(), is(equalTo(1)));
```
Comment (ronanrodrigo) [inline src/test/java/tasks/persistence/TaskGatewayTest.java]: Magic number :(

File: src/main/resources/static/app.js (hunk 1 above)
Comment (hcassus) [inline src/main/resources/static/app.js]: done

File: src/main/resources/static/app.js (hunk 2 above)
Comment (hcassus) [inline src/main/resources/static/app.js]: It stands for Controller, I'm gonna rename it too =)

File: src/main/resources/static/app.js (hunk 3 above)
Comment (hcassus) [inline src/main/resources/static/app.js]: `UNKNOWN` is my default value for unit. I haven't found a better way to enforce this initial state

File: src/test/java/tasks/api/TasksApiTest.java (hunk 5 above)
Comment (hcassus) [inline src/test/java/tasks/api/TasksApiTest.java]: in this case I am screwed by the framework... AFAIK, JSONPath can only iterate through these expressions. I'll try to look further to find better solutions

File: src/test/java/tasks/api/TasksApiTest.java (hunk 6 above)
Comment (hcassus) [inline src/test/java/tasks/api/TasksApiTest.java]: done

**Pull request merged**